fix(infra): TestHTTPStreamPlayer_APIPlayback_TTLExpiry のflaky解消（setPlaybackStatusの自動再生成削除＋テスト同期化）

## 概要

`internal/infra` の `TestHTTPStreamPlayer_APIPlayback_TTLExpiry` が main の CI でランダムに失敗する flaky テストになっている。原因は本番コード `setPlaybackStatus` の「レコードがなければ作る」ロジックと、テスト側でワーカー完了を待たずに `PrunePlaybacks` を呼んでいることの組み合わせ。

## 観測した失敗

- ワークフロー: build (run id `25294946419`, sha `ba83e0b`)
- 失敗箇所: `internal/infra/http_stream_player_test.go:3451`

```
--- FAIL: TestHTTPStreamPlayer_APIPlayback_TTLExpiry (0.00s)
    http_stream_player_test.go:3451: expected status=unknown after TTL expiry, got "completed"
FAIL	github.com/canpok1/vox-actor/internal/infra	1.910s
```

直前後のコミット (`562f1ed`, `c52b70d` 等) では同じテストが成功しており、再現性は低い。

## 原因

`internal/infra/http_stream_player.go` の `setPlaybackStatus` (L1024-1040) に以下の「自動再生成」ロジックがある:

```go
func (p *HTTPStreamPlayer) setPlaybackStatus(id string, status playbackStatus, reason string) {
	p.playbackMu.Lock()
	defer p.playbackMu.Unlock()
	if _, ok := p.playbacks[id]; !ok {
		p.playbacks[id] = &playbackStateRecord{createdAt: p.nowFunc()}
	}
	...
}
```

一方、テストは以下の流れで動作する:

1. `fixedTime = now - 2h` の状態で `POST /api/play` → `initPlayback` で `createdAt = now - 2h` のレコード作成、batchQueue に enqueue
2. ワーカーゴルーチンが非同期に処理開始（テストの `wav: []byte("RIFFx")` は不正な WAV のため worker は最終的に completed/failed のいずれかで `setPlaybackStatus` を呼ぶ）
3. テストが `fixedTime = time.Now()` に進めて `PrunePlaybacks()` を呼ぶ → 2h 前のレコードが削除される
4. **その直後にワーカーが `setPlaybackStatus` を呼ぶと、自動再生成ロジックが効いて `createdAt = 現在時刻 / status = completed` のレコードが復活する**
5. `GET /api/playback/<id>` が `unknown` ではなく `completed` を返してしまい失敗

CI ランナーが遅いとワーカーの `setPlaybackStatus` が PrunePlaybacks より後ろにずれ込み、再現する。

## 方針（C: テスト + プロダクション両方修正）

### 1. プロダクション側の修正

`setPlaybackStatus` の「無ければ作る」分岐を削除し、TTL で削除済み（または未登録）の id に対する更新は no-op にする。

```go
func (p *HTTPStreamPlayer) setPlaybackStatus(id string, status playbackStatus, reason string) {
	p.playbackMu.Lock()
	defer p.playbackMu.Unlock()
	state, ok := p.playbacks[id]
	if !ok {
		// pruned もしくは未登録。後追い更新は無視する。
		return
	}
	state.status = status
	state.reason = reason
	now := p.nowFunc().UnixMilli()
	if status == playbackStatusPlaying && state.startedAt == nil {
		state.startedAt = &now
	}
	if (status == playbackStatusCompleted || status == playbackStatusFailed) && state.finishedAt == nil {
		state.finishedAt = &now
	}
}
```

- `incrementCompletedClips` も同様に「無ければ作る」分岐は無いが、念のため挙動を再確認する。
- 自動再生成に依存している呼び出し元がないかを `grep` で事前確認する（`initPlayback` を経由しない `setPlaybackStatus` 直呼びの有無）。

### 2. テスト側の修正

`TestHTTPStreamPlayer_APIPlayback_TTLExpiry` で `PrunePlaybacks` を呼ぶ前にワーカー処理の完了を待つ。

- POST 後、`GET /api/playback/<id>` をポーリングして status が `completed` または `failed` になるまで待機（タイムアウト付き）
- その後で `fixedTime = time.Now()` に進めて `PrunePlaybacks()` を呼ぶ
- これにより、ワーカーが先に完了してから prune が走る順序を保証する

ヘルパーは同パッケージの既存 `helper_test.go` 等の有無を grep して配置を決める。

## 受け入れ条件

- [ ] `setPlaybackStatus` から自動再生成ロジックが削除され、pruned 済み id への更新は no-op になっている
- [ ] 削除した自動再生成に依存していた呼び出し元がないことを確認した（あれば併せて修正）
- [ ] `TestHTTPStreamPlayer_APIPlayback_TTLExpiry` がワーカー完了を待ってから `PrunePlaybacks` を呼ぶように修正されている
- [ ] `go test ./internal/infra -run TestHTTPStreamPlayer_APIPlayback_TTLExpiry -count=50` がローカルで全成功する
- [ ] `go test ./...` が成功する
- [ ] `gofmt -l .` / `golangci-lint run` の指摘がない

## 参考

- 失敗 run: https://github.com/canpok1/vox-actor/actions/runs/25294946419
- 対象テスト: `internal/infra/http_stream_player_test.go:3415-3453`
- 対象本番コード: `internal/infra/http_stream_player.go:1024-1040`
- 該当ファイル最終コミット: `947af4d`

---
🤖 Generated with [Claude Code](https://claude.ai/claude-code)

--- a/internal/infra/http_stream_player.go
+++ b/internal/infra/http_stream_player.go
@@ -1024,10 +1024,11 @@ func (p *HTTPStreamPlayer) initPlayback(id string, clipCount int) {
 func (p *HTTPStreamPlayer) setPlaybackStatus(id string, status playbackStatus, reason string) {
 	p.playbackMu.Lock()
 	defer p.playbackMu.Unlock()
-	if _, ok := p.playbacks[id]; !ok {
-		p.playbacks[id] = &playbackStateRecord{createdAt: p.nowFunc()}
+	state, ok := p.playbacks[id]
+	if !ok {
+		// pruned or unregistered; ignore late updates from workers.
+		return
 	}
-	state := p.playbacks[id]
 	state.status = status
 	state.reason = reason
 	now := p.nowFunc().UnixMilli()

--- a/internal/infra/http_stream_player_test.go
+++ b/internal/infra/http_stream_player_test.go
@@ -3412,6 +3412,30 @@ func TestHTTPStreamPlayer_Prefetch_SynthesizeFailureDuringPrefetchFails(t *testi
 	}
 }
 
+// waitForPlaybackTerminal polls GET /api/playback/<id> until status is "completed" or "failed",
+// or until the timeout is reached (test fails on timeout).
+func waitForPlaybackTerminal(t *testing.T, baseURL, playbackID string, timeout time.Duration) {
+	t.Helper()
+	deadline := time.Now().Add(timeout)
+	for time.Now().Before(deadline) {
+		resp, err := http.Get(baseURL + "/api/playback/" + playbackID)
+		if err != nil {
+			t.Fatalf("polling GET /api/playback: %v", err)
+		}
+		var result ViewerPlaybackResponse
+		if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
+			_ = resp.Body.Close()
+			t.Fatalf("decode polling response: %v", err)
+		}
+		_ = resp.Body.Close()
+		if result.Status == "completed" || result.Status == "failed" {
+			return
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+	t.Fatalf("timeout waiting for playback %s to reach terminal status", playbackID)
+}
+
 func TestHTTPStreamPlayer_APIPlayback_TTLExpiry(t *testing.T) {
 	t.Parallel()
 	stub := &testVoicevoxClient{wav: []byte("RIFFx")}
@@ -3433,6 +3457,10 @@ func TestHTTPStreamPlayer_APIPlayback_TTLExpiry(t *testing.T) {
 		t.Fatalf("decode POST response: %v", err)
 	}
 	_ = postResp.Body.Close()
+
+	// ワーカーが completed/failed に遷移するまで待機してから prune する。
+	// これにより、prune 後のワーカーによる自動再生成（修正前の挙動）が起きないことを保証する。
+	waitForPlaybackTerminal(t, "http://"+p.Addr(), playResult.PlaybackID, 2*time.Second)
 
 	// 現在時刻を "今" に進め、GC を走らせる（createdAt = 2h前 < cutoff = now - 1h）
 	fixedTime = time.Now()

